app/models/provider.rb

```diff
@@ -63,4 +63,20 @@ def refresh_ems(opts = {})
     end
     managers.flat_map { |manager| EmsRefresh.queue_refresh(manager, nil, opts) }
   end
+
+  def self.destroy_queue(ids)
+    find(Array.wrap(ids)).each(&:destroy_queue)
+  end
+
+  def destroy_queue
+    if managers.empty?
+      return destroy
+    end
+
+    _log.info("Queuing destroy of managers of provider: #{self.class.name} with id: #{id}")
+    managers.flat_map(&:destroy_queue)
+
+    _log.info("Queuing destroy of provider: #{self.class.name} with id: #{id}")
+    self.class._queue_task(:destroy_queue, id.to_miq_a, 15.seconds.from_now)
```

Comment:
No need for the `.to_miq_a`

Reply:
Or more specifically, it might be cleaner to use `Array.wrap` in the `_queue_task` method itself.

Reply:
done